libs/core/include/core/string/ends_with.hpp

```diff
@@ -0,0 +1,38 @@
+#pragma once
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include <string>
```

Comment:
Let's put the implementation in a cpp file. It worries me how much executable code we put into headers in the name of performance on no evidence, or just for convenience. Our rebuild times will thank us if we kick the habit.

Reply:
In general, I would agree, however in this case it is only used in the one file and we want to promote the inlining.

Reply:
Do we, though? How much time do we expect ledger to spend on constructing telemetry variables that an inlined function will make a difference?